spring-integration-core/src/main/java/org/springframework/integration/config/annotation/AbstractMethodAnnotationPostProcessor.java

```diff
@@ -106,67 +125,298 @@
-				this.logger.debug("Skipping endpoint creation; perhaps due to some '@Conditional' annotation.");
-				return null;
+	public void processBeanDefinition(String beanName, AnnotatedBeanDefinition beanDefinition,
+			List<Annotation> annotations) {
+
+		ResolvableType handlerBeanType = getHandlerBeanClass(beanDefinition);
+
+		String handlerBeanName = beanName;
+
+		BeanDefinition handlerBeanDefinition =
+				resolveHandlerBeanDefinition(beanName, beanDefinition, handlerBeanType, annotations);
+		MergedAnnotations mergedAnnotations = beanDefinition.getFactoryMethodMetadata().getAnnotations();
+
+		if (handlerBeanDefinition != null) {
+			if (handlerBeanDefinition != beanDefinition) {
+				Class<?> handlerBeanClass =
+						org.springframework.util.ClassUtils.resolveClassName(handlerBeanDefinition.getBeanClassName(),
+								this.beanFactory.getBeanClassLoader());
+
+				if (isClassIn(handlerBeanClass, Orderable.class, AbstractSimpleMessageHandlerFactoryBean.class)) {
+					mergedAnnotations.get(Order.class)
+							.getValue(AnnotationUtils.VALUE, String.class).
+							ifPresent((order) -> handlerBeanDefinition.getPropertyValues().add("order", order));
+				}
+
+				if (isClassIn(handlerBeanClass, AbstractMessageProducingHandler.class, AbstractMessageRouter.class,
+						AbstractSimpleMessageHandlerFactoryBean.class)) {
+
+					new BeanDefinitionPropertiesMapper(handlerBeanDefinition, annotations)
+							.setPropertyValue(SEND_TIMEOUT_ATTRIBUTE)
+							.setPropertyReference("outputChannel")
+							.setPropertyReference("defaultOutputChannel");
+				}
+
+				if (isClassIn(handlerBeanClass, AbstractMessageProducingHandler.class,
+						AbstractSimpleMessageHandlerFactoryBean.class)
+						&& !RouterFactoryBean.class.isAssignableFrom(handlerBeanClass)) {
+
+					applyAdviceChainIfAny(handlerBeanDefinition, annotations);
+				}
+
+				handlerBeanName = generateHandlerBeanName(beanName, mergedAnnotations);
+				this.definitionRegistry.registerBeanDefinition(handlerBeanName, handlerBeanDefinition);
+			}
+		}
+		else {
+			throw new BeanDefinitionValidationException(
+					"The messaging annotation processor for '" + this.annotationType +
+							"' cannot create a target handler bean. " +
+							"The bean definition with the problem is: " + beanName);
+		}
+
+		BeanDefinition endpointBeanDefinition =
+				createEndpointBeanDefinition(new BeanComponentDefinition(handlerBeanDefinition, handlerBeanName),
+						new BeanComponentDefinition(beanDefinition, beanName), annotations);
+
+		new BeanDefinitionPropertiesMapper(endpointBeanDefinition, annotations)
+				.setPropertyValue("autoStartup")
+				.setPropertyValue("phase");
+
+		Poller poller = MessagingAnnotationUtils.resolveAttribute(annotations, "poller", Poller.class);
+		Reactive reactive = MessagingAnnotationUtils.resolveAttribute(annotations, "reactive", Reactive.class);
+		Assert.state(reactive == null || poller == null, "The 'poller' and 'reactive' are mutually exclusive.");
```

Comment:
Either `"'poller' and 'reactive' are mutually exclusive."` or `"The 'poller' and 'reactive' attributes are mutually exclusive."` Also should include the bean name.